src/main/scala/io/movies/interpreters/RepositoryImpl.scala

```diff
@@ -0,0 +1,37 @@
+package io.movies.interpreters
+
+import cats.effect._
+import doobie._
+import doobie.hikari._
+import doobie.implicits._
+import io.movies.algebras.Repository
+import io.movies.model.{Movie, RegisteredMovie}
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+
+object RepositoryImpl {
+
+  //no se pasa el recurso para que no se crea/destruya cada vez. no es viable en el caso de conexiones a BBDD
+  def doobie[F[_]](transactor: HikariTransactor[F])(implicit S: Sync[F], logger: SelfAwareStructuredLogger[F]) : F[Repository[F]] =
+    S.pure(new Repository[F] {
```

Comment:
You don't need to lift the instance of `Repository` into the `F` context. You can return `Repository[F]` instead.

Reply:
That was already changed when the generator was not needed in the for comprehension because we added the logger